pydantic_ai_slim/pydantic_ai/tools.py

```diff
@@ -34,8 +32,16 @@
 """Type variable for agent dependencies."""
 
 
+class EndAgentRun(Exception):
```

Comment:
should probably move this into `exceptions.py` (or remove `exceptions.py` and put exceptions where they make most sense)

Reply:
also need to decide if this is explicitly public or explicitly private?

Reply:
I think best for it to be private to prevent it from being used in unexpected ways, given it's just trying to make `.stop_run` work.